src/modules/75-cf/pages/feature-flags-detail/flag-pipeline-tab/__tests__/FlagPipelineTab.test.tsx

```diff
@@ -0,0 +1,153 @@
+
+const renderComponent = (props: Partial<FlagPipelineTabProps> = {}): void => {
+  render(
+    <TestWrapper
+      path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/feature-flags/TEST_FLAG"
+      pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
+      queryParams={{ tab: 'flag_pipeline', activeEnvironment: 'TEST_ENV' }}
+    >
+      <FlagPipelineTab flagIdentifier="TEST_FLAG" {...props} />
+    </TestWrapper>
+  )
+}
+
+describe('FlagPipelineTab', () => {
+  const refetchMock = jest.fn()
+  beforeAll(() => {
+    jest.spyOn(cfServicesMock, 'useGetAvailableFeaturePipelines').mockReturnValue({
+      data: {
+        availablePipelines: mockAvailablePipelines
+      },
+      loading: false,
+      refetch: refetchMock
+    } as any)
+  })
+
+  test('it should let user select a flag pipeline when non configured', async () => {
+    renderComponent()
+
+    // assert empty state initially
+    const addFlagPipelineButton = screen.getByRole('button', { name: 'cf.featureFlags.flagPipeline.noDataButtonText' })
+    expect(screen.getByText('cf.featureFlags.flagPipeline.noDataMessage')).toBeInTheDocument()
+    expect(addFlagPipelineButton).toBeInTheDocument()
+    expect(screen.getByText('cf.featureFlags.flagPipeline.noDataDescription')).toBeInTheDocument()
+
+    // assert drawer opens with correct text. Save button hidden initially
+    addFlagPipelineButton.click()
+    expect(screen.getByText('cf.featureFlags.flagPipeline.drawerTitle')).toBeInTheDocument()
+    expect(screen.getByText('cf.featureFlags.flagPipeline.drawerDescription')).toBeInTheDocument()
+    expect(screen.getByText('cf.featureFlags.flagPipeline.drawerEnvText')).toBeInTheDocument()
+    expect(
+      screen.queryByRole('button', { name: 'cf.featureFlags.flagPipeline.drawerButtonText' })
+    ).not.toBeInTheDocument()
+
+    // assert correct number of pipelines and pipeline content is displayed
+    await waitFor(() => expect(screen.getAllByRole('listitem')).toHaveLength(mockAvailablePipelines.length))
+    await waitFor(() =>
+      expect(screen.getAllByRole('listitem')[0]).toHaveTextContent('Pipeline 1Pipeline 1 description')
+    )
+
+    // select a pipeline
+    const itemToSelect = screen.getAllByRole('listitem')[4]
+    userEvent.click(itemToSelect)
+    expect(itemToSelect).toHaveClass('cardActive')
+
+    const savePipelineButton = screen.getByRole('button', { name: 'cf.featureFlags.flagPipeline.drawerButtonText' })
+    expect(savePipelineButton).toBeInTheDocument()
+  })
+
+  test('it should let user search for a flag pipeline', async () => {
+    renderComponent()
+    const addFlagPipelineButton = screen.getByRole('button', {
+      name: 'cf.featureFlags.flagPipeline.noDataButtonText'
+    })
+    addFlagPipelineButton.click()
+
+    const searchInput = screen.getByPlaceholderText('Search')
+    expect(searchInput).toBeInTheDocument()
+
+    userEvent.type(searchInput, 'pipeline 1')
```

Comment:
There's a built-in delay between key presses, so every character you `type` takes extra time. If you pass `{ allAtOnce: true }` as the 3rd parameter it'll kinda act like it was pasted and there will be no delay. It can be the difference between a stable test and a test that times out and becomes flaky.
```suggestion
    userEvent.type(searchInput, 'pipeline 1', { allAtOnce: true })
```

Reply:
fixed